Oraimunka24015004/src/AlgorithmicRecap.java

```diff
@@ -0,0 +1,188 @@
+            for (int j = i + 1; j < text.length(); j++) {
+                char tempChar = text.charAt(j);
+                if (tempChar == character) {
+                    counter++;
+                }
+            }
+            if (counter == 0 && !blacklist.contains(character)) {
+                result.append((character));
+            } else {
+                blacklist.add(character);
+                counter = 0;
+            }
+        }
+        return result;
+    }
+
+    boolean taskEightSameAmountOfAb(String text, String text2) {
+        return (taskEightMethod('a', text) == taskEightMethod('a', text2))
+                && taskEightMethod('b', text) == taskEightMethod('b', text2);
+    }
+
+    private int taskEightMethod(char getChar, String text) {
+        int counter = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (character == getChar) {
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    boolean taskNineArrayContains(int[] numbers, int number) {
+        Arrays.sort(numbers);
+        for (int numb : numbers) {
+            if (number == numb) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    StringBuilder taskTenWhichString(List<String> texts, char character) {
+        StringBuilder result = new StringBuilder();
+        for (String text : texts) {
+            for (int i = 0; i < text.length(); i++) {
+                char getChar = text.charAt(i);
+                if (getChar == character) {
+                    result.append(text + ",");
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    List<Integer> taskElevenReturnSum(int[] numbers) {
+        List<Integer> list = new ArrayList<>();
+        int counter = 0;
+        for (int i = 0; i < numbers.length; i++) {
+            if (i == 0) {
+                list.add(numbers[i]);
+            } else {
+                list.add(list.get(counter) + numbers[i]);
+                counter++;
+            }
+        }
+        return list;
+    }
+
+    boolean taskTwelveIsPrime(int number1, int number2) {
+        return taskTwelveMethod(number1) && taskTwelveMethod(number2);
+    }
+
+    private boolean taskTwelveMethod(int number) {
+        if (number < 2) {
+            return false;
+        }
+        for (int i = 2; i <= Math.sqrt(number); i++) {
+            if (number % i == 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    List taskThirteenFindMatthewKnight(String[][] arrays) {
+        List<String> routes = new ArrayList<>();
+        for (int i = 0; i < arrays.length; i++) {
+            for (int j = 0; j <= 1; j++) {
+                if (i == 0 && (j == 0 || j == 1)) {
+                    routes.add(arrays[i][j]);
+                }
+                if (arrays[i][0].equals("BRA")) {
```

Comment:
And what if I start with another country, not `BRA`? Don't solve the problem with specific inputs, but try to tackle the generic problem instead. Your algorithm will fail in any other case apart from whether the guy is starting his journey from Brazil